src/hooks/useTitleChanges.js

```diff
@@ -0,0 +1,43 @@
+import { useMemo, useEffect, useCallback } from 'react'
+import { getAppFullName } from 'lib/utils.js'
+
+function useTitleChanges({ noteId, title, setTitle, serviceClient }) {
+  // Title change because of a local change from the editor
+  const onLocalTitleChange = useCallback(
+    serviceClient
+      ? e => {
+          const newTitle = e.target.value
+          const modifiedTitle = newTitle
+          if (title != modifiedTitle) {
+            setTitle(modifiedTitle)
+            serviceClient.setTitle(noteId, modifiedTitle)
+          }
+        }
+      : () => {},
+    [noteId, title, setTitle, serviceClient]
+  )
+  // Title change because of a remote change from the stack
+  useMemo(
```

Comment:
This should be a `useEffect`, right? `useMemo` is for memoized values, but this function doesn't return anything.